src/viam/examples/modules/mlmodelservice_tflite.cpp

```diff
@@ -0,0 +1,759 @@
+            }
+            case kTfLiteUInt64: {
+                return make_tensor_view_t_<std::uint64_t>(info, tflite_tensor);
+            }
+            case kTfLiteFloat32: {
+                return make_tensor_view_t_<float>(info, tflite_tensor);
+            }
+            case kTfLiteFloat64: {
+                return make_tensor_view_t_<double>(info, tflite_tensor);
+            }
+            default: {
+                std::ostringstream buffer;
+                buffer << service_name
+                       << ": Model returned unsupported tflite data type: " << tflite_tensor_type;
+                throw std::invalid_argument(buffer.str());
+            }
+        }
+    }
+
+    // The type specific version of the above function, it just
+    // reinterpret_casts the tensor buffer into an MLModelService
+    // tensor view and applies the necessary shape info.
+    template <typename T>
+    MLModelService::tensor_views make_tensor_view_t_(const MLModelService::tensor_info& info,
+                                                     const TfLiteTensor* const tflite_tensor) {
+        const auto* const tensor_data = reinterpret_cast<const T*>(TfLiteTensorData(tflite_tensor));
+        const auto tensor_size_bytes = TfLiteTensorByteSize(tflite_tensor);
+        const auto tensor_size_t = tensor_size_bytes / sizeof(T);
+        // TODO: We are just feeding back out what we cached in the
+        // metadata for shape. Should this instead be re-querying the
+        // output tensor NumDims / DimN after each invocation in case
+        // the shape is dynamic? The possibility of a dynamically
+        // sized extent is why we represent the dimensions as signed
+        // quantities in the tensor metadata. But an actual tensor has
+        // a real extent. How would tflite ever communicate that to us
+        // differently given that we use the same API to obtain
+        // metadata as we would here?
+        std::vector<std::size_t> shape;
+        shape.reserve(info.shape.size());
+        for (const auto s : info.shape) {
+            shape.push_back(static_cast<std::size_t>(s));
+        }
+        return MLModelService::make_tensor_view(tensor_data, tensor_size_t, std::move(shape));
+    }
+
+    // The mutex and condition variable needed to track our state
+    // across concurrent reconfiguration and invocation.
+    std::mutex state_lock_;
+    std::condition_variable state_ready_;
+    std::shared_ptr<state> state_;
+};
+
+int serve(const std::string& socket_path) try {
+    // Block the signals we intend to wait for synchronously.
+    sigset_t sigset;
+    sigemptyset(&sigset);
+    sigaddset(&sigset, SIGINT);
+    sigaddset(&sigset, SIGTERM);
+    pthread_sigmask(SIG_BLOCK, &sigset, NULL);
+
+    // Create a new model registration for the service.
+    auto module_registration = std::make_shared<vsdk::ModelRegistration>(
+        // TODO: Document me.
```

Comment:
@stuqdog - I'd like some more details on what this field does / means.

Reply:
Per offline conversation, I think it's an artifact, model/resourceAPI should give all that's necessary.